grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy

```diff
@@ -776,80 +782,84 @@ class ProductAvailabilityService {
         List searchTerms = (command?.searchTerms ? Arrays.asList(command?.searchTerms?.split(" ")) : null)
 
         // Only search if there are search terms otherwise the list of product IDs includes all products
-        def innerProductIds = !searchTerms ? [] : Product.createCriteria().list {
-            eq("active", true)
-            projections {
-                property 'id'
-            }
-            and {
-                searchTerms.each { searchTerm ->
-                    or {
-                        ilike("name", "%" + searchTerm + "%")
-                        synonyms {
-                            and {
-                                ilike("name", "%" + searchTerm + "%")
-                                eq("synonymTypeCode", SynonymTypeCode.DISPLAY_NAME)
-                            }
-                        }
-                        inventoryItems {
-                            ilike("lotNumber", "%" + searchTerm + "%")
-                        }
-                    }
-                }
-            }
-        }.unique()
+        def innerProductIds = searchTerms ?
+                productService.searchProducts(searchTerms.toArray(), [])?.collect { it.id } : []
+
+        // Retrieve all product types with SEARCHABLE and SEARCHABLE_NO_STOCK activity codes
+        String productTypeQuery = "select pt from ProductType pt left join pt.supportedActivities sa where sa=:productActivityCode"
```

Comment:
nice job, this is what I needed and didn't know how to get it to be able to use it in the `createCriteria`